File: ios/MoveHome/App/MoveHomeViewModel.swift
import Foundation
import Combine

/**
 * Thin UI adapter over CaptureManager + CaptureState.
 *
 * DELETABLE: Once the Action Button intent and home/lock screen widget are the
 * primary interaction points, this ViewModel, ContentView, and MoveHomeApp
 * can all be removed. CaptureManager + CaptureState are the permanent core.
 */
class MoveHomeViewModel: ObservableObject {

    @Published var isCapturing: Bool    = CaptureState.shared.isCapturing
    @Published var lastGesture: String? = CaptureState.shared.lastGesture

    private var cancellables = Set<AnyCancellable>()

    init() {
        CaptureManager.shared.onGestureDetected = { [weak self] gesture in
            self?.lastGesture = gesture
        }
        CaptureState.shared.$isCapturing
            .receive(on: DispatchQueue.main)
            .assign(to: &$isCapturing)
        CaptureState.shared.$lastGesture
            .receive(on: DispatchQueue.main)
            .assign(to: &$lastGesture)
    }

    func toggleCapture() {
        CaptureManager.shared.toggle()
    }

    func setupBackTap() {
        NotificationCenter.default.addObserver(
            forName: .init("MoveHomeBackTap"),
            object: nil,
            queue: .main
        ) { [weak self] _ in self?.toggleCapture() }
    }
}

File: android/src/main/kotlin/com/movehome/android/MoveHomeViewModel.kt
package com.movehome.android

import android.app.Application
import android.content.Intent
import androidx.lifecycle.AndroidViewModel
import com.movehome.android.service.MoveHomeForegroundService
import com.movehome.shared.model.GestureType
import kotlinx.coroutines.flow.StateFlow

/**
 * Thin UI adapter over MoveHomeForegroundService + CaptureState.
 *
 * DELETABLE: Once the Quick Settings tile and home screen widget are the
 * primary interaction points, this ViewModel and MainActivity can be removed.
 * MoveHomeForegroundService + CaptureState are the permanent core.
 */
class MoveHomeViewModel(private val app: Application) : AndroidViewModel(app) {

    val isCapturing: StateFlow<Boolean>    = CaptureState.isCapturing
    val lastGesture: StateFlow<GestureType?> = CaptureState.lastGesture

    fun toggleCapture() = sendAction(MoveHomeForegroundService.ACTION_TOGGLE)
    fun onVolumeDown()  = sendAction(MoveHomeForegroundService.ACTION_TOGGLE)

    private fun sendAction(action: String) {
        app.startService(
            Intent(app, MoveHomeForegroundService::class.java).apply { this.action = action }
        )
    }
}
